File: BE/XinMall/src/main/java/com/example/xinmall/entity/system/Banner.java
package com.example.xinmall.entity.system;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("banner")
public class Banner {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String image;

    private String title;

    private String link;

    private Integer sort;

    private Integer status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: FE/src/api/banner.ts
import { http } from '@/utils/http';

export interface BannerItem {
  id: number;
  image: string;
  title: string;
  link: string;
  sort: number;
  status: number;
  createdAt: string;
  updatedAt: string;
}

export const bannerApi = {
  getBannerList: () => {
    return http<BannerItem[]>({
      url: '/banner/list',
      method: 'GET'
    });
  }
};
